src/main/java/world/bentobox/warps/listeners/WarpSignsListener.java

```diff
@@ -44,6 +43,29 @@ public WarpSignsListener(Warp addon) {
         this.plugin = addon.getPlugin();
     }
 
+    @EventHandler(priority = EventPriority.NORMAL)
+    public void onChunkLoad(ChunkLoadEvent event) {
+        boolean changed = false;
+        Iterator<Map.Entry<UUID, Location>> iterator =
+                addon.getWarpSignsManager().getWarpMap(event.getWorld()).entrySet().iterator();
+        while (iterator.hasNext()) {
+            Map.Entry<UUID, Location> entry = iterator.next();
+            UUID uuid = entry.getKey();
+            Location location = entry.getValue();
+            if (event.getChunk().getX() == location.getBlockX() >> 4
+                    && event.getChunk().getZ() == location.getBlockZ() >> 4
+                    && !Tag.SIGNS.isTagged(location.getBlock().getType())) {
+                iterator.remove();
```

Comment:
I wonder if a CME is being thrown because of multi-threaded chunk loading? 
You might need to build a list of wrong warps and then iterate over that to remove them.

Reply:
is ChunkLoadEvent Async now? O_o

Reply:
I double checked and ChunkLoadEvent is always called sync